storops_test/unity/resource/test_snap_schedule.py

```diff
@@ -0,0 +1,443 @@
+         'is_auto_delete': None, 'retention_time': None, 'access_type': None,
+         'error_msg': 'days_of_week of UnitySnapScheduleRule every_week rule '
+                      'should be a list.'},
+        {'days_of_week': [], 'hour': None, 'minute': None,
+         'is_auto_delete': None, 'retention_time': None, 'access_type': None,
+         'error_msg': 'days_of_week of UnitySnapScheduleRule every_week rule '
+                      'cannot be empty list.'}
+    )
+    def test_every_week_value_error(self, kwargs):
+        days_of_week = kwargs.pop('days_of_week')
+        error_msg = kwargs.pop('error_msg')
+        kwargs = {key: val for key, val in kwargs.items() if val is not None}
+        assert_that(
+            calling(UnitySnapScheduleRule.every_week).with_args(days_of_week,
+                                                                **kwargs),
+            raises(ValueError, error_msg))
+
+    @ddt.data(
+        {'days_of_month': [10, 20], 'hour': None, 'minute': None,
+         'is_auto_delete': None, 'retention_time': None, 'access_type': None,
+         'expected':
+             {'type': ScheduleTypeEnum.NTH_DAYOFMONTH_AT_HHMM,
+              'minute': 0, 'hours': [0],
+              'daysOfWeek': None,
+              'daysOfMonth': [10, 20],
+              'interval': None, 'isAutoDelete': None, 'retentionTime': None,
+              'accessType': None}},
+        {'days_of_month': [10, 20], 'hour': 12, 'minute': 30,
+         'is_auto_delete': True, 'retention_time': 3600,
+         'access_type': FilesystemSnapAccessTypeEnum.CHECKPOINT,
+         'expected':
+             {'type': ScheduleTypeEnum.NTH_DAYOFMONTH_AT_HHMM,
+              'minute': 30, 'hours': [12],
+              'daysOfWeek': None,
+              'daysOfMonth': [10, 20],
+              'interval': None, 'isAutoDelete': True, 'retentionTime': 3600,
+              'accessType': FilesystemSnapAccessTypeEnum.CHECKPOINT}},
+    )
+    def test_every_month(self, kwargs):
+        days_of_month = kwargs.pop('days_of_month')
+        expected = kwargs.pop('expected')
+        kwargs = {key: val for key, val in kwargs.items() if val is not None}
+        assert_that(UnitySnapScheduleRule.every_month(days_of_month,
+                                                      **kwargs),
+                    equal_to(expected))
+
+    @ddt.data(
+        {'days_of_month': 10, 'hour': None, 'minute': None,
+         'is_auto_delete': None, 'retention_time': None, 'access_type': None,
+         'error_msg': 'days_of_month of UnitySnapScheduleRule every_month '
+                      'rule should be a list.'},
+        {'days_of_month': [], 'hour': None, 'minute': None,
+         'is_auto_delete': None, 'retention_time': None, 'access_type': None,
+         'error_msg': 'days_of_month of UnitySnapScheduleRule every_month '
+                      'rule cannot be empty list.'}
+    )
+    def test_every_month_exception(self, kwargs):
```

Comment:
Why not follow the naming rule of the previous negative cases: test_xxx_value_error

Reply:
Changed it to `test_every_month_value_error`.